src/transports/sse.ts

```diff
@@ -110,6 +111,15 @@ export class SseTransport<T extends SSETransportGenerics> extends StreamingTrans
           const results = listener.parseResponse(e).map((r) => {
             const partialResponse = r.response as PartialSuccessfulResponse<T['Response']>
             const result = r as TimestampedProviderResult<T>
+            if (partialResponse.timestamps?.providerIndicatedTimeUnixMs !== undefined) {
+              const timestampValidator = validator.responseTimestamp()
+              const error = timestampValidator(
+                partialResponse.timestamps?.providerIndicatedTimeUnixMs,
+              )
+              if (error) {
+                logger.warn(`Provider indicated time is invalid: ${error}`)
+              }
+            }
```

Comment:
Instead of doing it in each transport, let's add this check to the response cache write